src/witness_generator/bit_constraints.rs

```diff
@@ -45,21 +45,27 @@ impl BitConstraint {
     /// Returns the conjunction of this constraint and the other.
     pub fn conjunction(self, other: &BitConstraint) -> BitConstraint {
         BitConstraint {
-            mask: self.mask & other.mask.clone(),
+            mask: self.mask & other.mask,
         }
     }
 
     /// The bit constraint of an integer multiple of an expression.
     /// TODO this assumes goldilocks
     pub fn multiple(&self, factor: AbstractNumberType) -> Option<BitConstraint> {
-        if factor.clone() * self.mask.clone() >= get_field_mod() {
+        if factor
+            .clone()
+            .to_degree()
+            .checked_mul(self.mask)
```

Comment:
Shouldn't this be arbitrary precision instead? Or maybe u128?

Reply:
Changed to u128